src/tfr_aruco/src/aruco_action_server.cpp

```diff
@@ -97,33 +98,37 @@ class TFR_Aruco {
         
         cv::aruco::detectMarkers(imageHolder->image, dictionary, markerCorners, markerIds, params);
     
-    
         // get individual marker poses
         cv::Mat cameraMatrix = cv::Mat(cameraModel.fullIntrinsicMatrix()).clone();
         cv::Mat distCoeffs = cameraModel.distortionCoeffs().clone();
         
         std::vector< cv::Vec3d > rvecs, tvecs;
         cv::aruco::estimatePoseSingleMarkers(markerCorners, 0.05, cameraMatrix, distCoeffs, rvecs, tvecs);
-    
-    
-    
         
         cv::Vec3d boardRotVec, boardTransVec;
         int markersDetected = cv::aruco::estimatePoseBoard(markerCorners, markerIds, board, cameraMatrix, distCoeffs, boardRotVec, boardTransVec);
         
         tfr_msgs::ArucoResult result;
         result.number_found = markersDetected;
-        result.position = std::vector<float>(3, 0); // 3 members initialized to 0
-        result.rotation = std::vector<float>(3, 0); // 3 members initialized to 0
-        if(markersDetected != 0) 
+        if (result.number_found > 0)
         {
-            result.position[0] = boardTransVec[0];
-            result.position[1] = boardTransVec[1];
-            result.position[2] = boardTransVec[2];
-    
-            result.rotation[0] = boardRotVec[0];
-            result.rotation[1] = boardRotVec[1];
-            result.rotation[2] = boardRotVec[2];
+            result.relative_pose.header.stamp = ros::Time::now();
+            result.relative_pose.header.frame_id = goal->image.header.frame_id;
+            /*
+             *  also the coordinate axist for the aruco are in a different
+             *  coordinate system and are rotated here.
+             * */
+            result.relative_pose.pose.position.x = boardTransVec[2];
+            result.relative_pose.pose.position.y = boardTransVec[0];
+            result.relative_pose.pose.position.z = boardTransVec[1];
+            //let tf do the euler angle -> quaternion math
+            tf2::Quaternion rotated{};
+            //change rotated perspective RPY aruco output to ros coordinate system (2d)
+            rotated.setRPY(boardRotVec[2], boardRotVec[0], 3.1415 + boardRotVec[1]);
```

Comment:
Isn't there a pi constant somewhere? Not a huge issue but might be nicer to use that

Reply:
I can do that as a variable. Surprisingly a constant for pi is not in the `c++` standard, and ros doesn't have one either.

Reply:
There is one, but it's not portable.